src/core/hyper-ref/index.js

```diff
@@ -389,16 +389,33 @@ export default class HyperRef extends PureComponent<Props, State> {
     if (!behaviors.length) {
       return children;
     }
-    const visibleHandlers = behaviors.map(behaviorElement =>
-      this.createActionHandler(behaviorElement, this.props.onUpdate),
-    );
     const onVisible = () => {
-      visibleHandlers.forEach(h => h(this.props.element));
+      // We don't want to use the cached `behaviors` list here because
+      // the DOM might have been mutated since.
+      this.getBehaviorElements(TRIGGERS.VISIBLE)
+        .map(behaviorElement =>
+          this.createActionHandler(behaviorElement, this.props.onUpdate),
+        )
+        .forEach(h => h(this.props.element));
     };
 
+    // If element does not have an `id` attribute, generate a pseudo id from list of attributes
+    // This is necessary to indicate to the VisibilityDetectingView that the element has changed
+    // and the internal state needs to be reset.
+    const id =
```

Comment:
I guess an alternative would be to take a hash of the stringified content of the element. I think this should work for now, but we should look out for any potential issues down the line.